Description: clarify warning message about PIC/PIE
Origin: upstream, https://code.qt.io/cgit/qt/qtbase.git/commit/?id=e2265965083343aa
Last-Update: 2020-07-08

--- a/src/corelib/global/qglobal.h
+++ b/src/corelib/global/qglobal.h
@@ -1280,7 +1280,7 @@ Q_CORE_EXPORT int qrand();
 #if !defined(QT_BOOTSTRAPPED) && defined(QT_REDUCE_RELOCATIONS) && defined(__ELF__) && \
     (!defined(__PIC__) || (defined(__PIE__) && defined(Q_CC_GNU) && Q_CC_GNU >= 500))
 #  error "You must build your code with position independent code if Qt was built with -reduce-relocations. "\
-         "Compile your code with -fPIC (-fPIE is not enough)."
+         "Compile your code with -fPIC (and not with -fPIE)."
 #endif
 
 namespace QtPrivate {
