Description: xcb: round down the scale factor for values < N + 0.8
Author: Mathieu Velten <matmaul@gmail.com>
Forwarded: https://codereview.qt-project.org/231719
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=884956
Last-Update: 2018-09-19

--- a/src/plugins/platforms/xcb/qxcbscreen.cpp
+++ b/src/plugins/platforms/xcb/qxcbscreen.cpp
@@ -753,7 +753,12 @@ void QXcbScreen::updateGeometry(const QR
         m_sizeMillimeters = sizeInMillimeters(geometry.size(), m_virtualDesktop->dpi());
 
     qreal dpi = geometry.width() / physicalSize().width() * qreal(25.4);
-    m_pixelDensity = qMax(1, qRound(dpi/96));
+    qreal rawFactor = dpi/96;
+    int roundedFactor = qFloor(rawFactor);
+    // Round up for .8 and higher. This favors "small UI" over "large UI".
+    if (rawFactor - roundedFactor >= 0.8)
+        roundedFactor = qCeil(rawFactor);
+    m_pixelDensity = qMax(1, roundedFactor);
     m_geometry = geometry;
     m_availableGeometry = geometry & m_virtualDesktop->workArea();
     QWindowSystemInterface::handleScreenGeometryChange(QPlatformScreen::screen(), m_geometry, m_availableGeometry);
