Description: fix crash in QWindow::mapFromGlobal() on High DPI
Origin: upstream, https://code.qt.io/cgit/qt/qtbase.git/commit/?id=6d61b10f65cd276e
Last-Update: 2019-07-03

--- a/src/gui/kernel/qhighdpiscaling.cpp
+++ b/src/gui/kernel/qhighdpiscaling.cpp
@@ -400,7 +400,7 @@ QPoint QHighDpiScaling::mapPositionToGlo
 QPoint QHighDpiScaling::mapPositionFromGlobal(const QPoint &pos, const QPoint &windowGlobalPosition, const QWindow *window)
 {
     QPoint windowPosCandidate = pos - windowGlobalPosition;
-    if (QGuiApplicationPrivate::screen_list.size() <= 1)
+    if (QGuiApplicationPrivate::screen_list.size() <= 1 || window->handle() == nullptr)
         return windowPosCandidate;
 
     // Device independent global (screen) space may discontiguous when high-dpi scaling
