Description: fix installation of resources for example sources
Origin: upstream, https://code.qt.io/cgit/qt/qtbase.git/commit/?id=22d3eeebb0a011a6
Last-Update: 2018-02-22

--- a/mkspecs/features/qt_example_installs.prf
+++ b/mkspecs/features/qt_example_installs.prf
@@ -44,7 +44,7 @@
 for(ex, EXAMPLE_FILES): \
     sourcefiles += $$files($$absolute_path($$ex, $$_PRO_FILE_PWD_))
 for(res, RESOURCES) {
-    !contains(res, \\.qrc$): \
+    !contains(res, .*\\.qrc): \
         next()
     rfile = $$absolute_path($$res, $$_PRO_FILE_PWD_)
     rpath = $$dirname(rfile)
